Clean up code for internal CSS values

nt1m opened the pull request:
#### 8c8cfbf1f0f0640439a7a9d5d2972c972b7063a7
<pre>
Clean up code for internal CSS values
<a href="https://bugs.webkit.org/show_bug.cgi?id=267520">https://bugs.webkit.org/show_bug.cgi?id=267520</a>
<a href="https://rdar.apple.com/121019340">rdar://121019340</a>

Reviewed by Darin Adler.

Right now, if you want to make a certain value internal, you have to do 2 changes:
1) Add &quot;status&quot;: &quot;internal&quot; to CSSProperties.json
2) Change isValueAllowedInMode() to include your value

This is error prone, as it&apos;s easy to think that the first step is sufficient.

With this change:
- the first step will be sufficient to make a value UA-sheet only
- isValueAllowedInMode is now named isColorKeywordAllowedInMode as it&apos;s now only used for color parsing.
- we now validate that all values starting with &quot;-internal-&quot; have &quot;status&quot;: &quot;internal&quot; in CSSProperties.json

Also use isUASheetBehavior whenever possible.

* Source/WebCore/css/parser/CSSParserContext.cpp:
(WebCore::CSSParserContext::CSSParserContext):
* Source/WebCore/css/parser/CSSParserFastPaths.cpp:
(WebCore::parseColor):
* Source/WebCore/css/parser/CSSParserIdioms.cpp:
(WebCore::isColorKeywordAllowedInMode):
(WebCore::isValueAllowedInMode): Deleted.
* Source/WebCore/css/parser/CSSParserIdioms.h:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::LengthRawKnownTokenTypeDimensionConsumer::consume):
(WebCore::CSSPropertyParserHelpers::consumeColorWorkerSafe):
(WebCore::CSSPropertyParserHelpers::consumeColor):
(WebCore::CSSPropertyParserHelpers::consumeCounterStyleNameInPrelude):
(WebCore::CSSPropertyParserHelpers::consumeDisplay):
* Source/WebCore/css/parser/CSSSelectorParser.cpp:
(WebCore::extractCompoundFlags):
(WebCore::CSSSelectorParser::consumeSimpleSelector):
(WebCore::CSSSelectorParser::splitCompoundAtImplicitShadowCrossingCombinator):
* Source/WebCore/css/process-css-properties.py:
Directly check for UASheetMode and add validation for `-internal-` values that they use correct &quot;status&quot; field.

* Source/WebCore/css/process-css-pseudo-selectors.py:
The .join() case also works fine for when there&apos;s only one condition.

Canonical link: <a href="https://commits.webkit.org/273071@main">https://commits.webkit.org/273071@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/41f5d512ae401b6df2b3b56505aeb640062e7ee7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/34151 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/12955 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/36132 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/36833 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/30946 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/35229 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/15339 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/10088 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/29980 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/34657 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/10958 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/30464 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/9567 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/9671 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/30498 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/38123 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/31022 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/30812 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/35768 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/9796 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/7683 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/33680 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/11583 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7866 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/10359 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/10613 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/css/parser/CSSParserIdioms.h:
```diff
@@ -60,7 +60,7 @@ bool isNameCodePoint(CharacterType c)
 }
 
-bool isValueAllowedInMode(unsigned short, CSSParserMode);
+bool isColorValueAllowedInMode(unsigned short, CSSParserMode);
```
Seems to me this argument needs a name or should be `CSSValueID` instead of `unsigned short`.

nt1m replied:
Fixed

nt1m committed:
Clean up code for internal CSS values
https://bugs.webkit.org/show_bug.cgi?id=267520
rdar://121019340

Reviewed by Darin Adler.

Right now, if you want to make a certain value internal, you have to do 2 changes:
1) Add "status": "internal" to CSSProperties.json
2) Change isValueAllowedInMode() to include your value

This is error prone, as it's easy to think that the first step is sufficient.

With this change:
- the first step will be sufficient to make a value UA-sheet only
- isValueAllowedInMode is now named isColorKeywordAllowedInMode as it's now only used for color parsing.
- we now validate that all values starting with "-internal-" have "status": "internal" in CSSProperties.json

Also use isUASheetBehavior whenever possible.

* Source/WebCore/css/parser/CSSParserContext.cpp:
(WebCore::CSSParserContext::CSSParserContext):
* Source/WebCore/css/parser/CSSParserFastPaths.cpp:
(WebCore::parseColor):
* Source/WebCore/css/parser/CSSParserIdioms.cpp:
(WebCore::isColorKeywordAllowedInMode):
(WebCore::isValueAllowedInMode): Deleted.
* Source/WebCore/css/parser/CSSParserIdioms.h:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::LengthRawKnownTokenTypeDimensionConsumer::consume):
(WebCore::CSSPropertyParserHelpers::consumeColorWorkerSafe):
(WebCore::CSSPropertyParserHelpers::consumeColor):
(WebCore::CSSPropertyParserHelpers::consumeCounterStyleNameInPrelude):
(WebCore::CSSPropertyParserHelpers::consumeDisplay):
* Source/WebCore/css/parser/CSSSelectorParser.cpp:
(WebCore::extractCompoundFlags):
(WebCore::CSSSelectorParser::consumeSimpleSelector):
(WebCore::CSSSelectorParser::splitCompoundAtImplicitShadowCrossingCombinator):
* Source/WebCore/css/process-css-properties.py:
Directly check for UASheetMode and add validation for `-internal-` values that they use correct "status" field.

* Source/WebCore/css/process-css-pseudo-selectors.py:
The .join() case also works fine for when there's only one condition.

Canonical link: https://commits.webkit.org/273071@main
